### PR TITLE
Bug fixes issue #27 - added handling for mime types

### LINKED ISSUE #27: Serve all pluginfile with correct mimetype
When loading in some envs the js is served as html and so won't load:

console message  ' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled

### PR DESCRIPTION
**Pull request by glepod**
Closes #27

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -156,6 +158,14 @@ function scormremote_pluginfile($course, $cm, $context, $filearea, $args, $force
         return false;
     }
 
+    // Handle special case mime types.
+    $mimetype = optional_param('mimetype', null, PARAM_RAW);
```
Comment (brendanheywood) [inline lib.php]: I would not make this a query param as it potentially opens security holes

I'd just make it emit the header just before calling send_file or similar

File: lib.php [hunk 2]
```diff
@@ -27,6 +27,8 @@
 use mod_scormremote\tier;
 use mod_scormremote\utils;
 
+// Set mime type values for serving content.
+const MIME_TYPE_JAVASCRIPT = 'text/javascript';
```
Comment (brendanheywood) [inline lib.php]: This isn't likely to change so I'd just hard code this in the call to header

File: lib.php [hunk 3]
```diff
@@ -239,9 +249,11 @@ function scormremote_pluginfile($course, $cm, $context, $filearea, $args, $force
             'layer3.js'
         );
 
+//            'jssource'         => $jssource . "?lms_origin={$origin}&student_id={$username}&student_name={$fullname}&mime='text/javascript'",
```
Comment (brendanheywood) [inline lib.php]: this will all get moved around anyway but generally don't leave commented code in

File: lib.php (hunk 1 above)
Comment (glepod) [reply lib.php]: It had to be conditional as scormremote_pluginfile serves html files as well.
I could change the condition to check the extension .js and emit the header on that case.
Not sure about security issue, there are exposed params already including the file type (.js extension)

**Pull request merged**